src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -61,6 +61,19 @@
  * <pre>
  * &lt;module name=&quot;NoWhitespaceBefore&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * int a;
+ * a ++; // violation, '++' is preceded with whitespace
+ * a++; // OK
+ * int foo
+ * ; // violation, linebreak is not allowed by default
+ * int bar; // OK
+ * int baz ; // violation, ';' is preceded with whitespace
+ * for (int i = 0 ; i &lt; 5; i++) {}
+ *            // ^ violation, ';' is preceded with whitespace
```

Comment:
comment "violation" should be on same line as actual violation.
comment that exaplains why it is violation can be on other line.

Reply:
done